clone/cmd/sumdbverify/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.22.8-alpine3.20@sha256:f56a8a4a1aea41bc4694728b69c219af1523aea15690cbbed82dc9bac81e6603 AS builder
+FROM golang:1.23.6-alpine3.20@sha256:22caeb4deced0138cb4ae154db260b22d1b2ef893dde7f84415b619beae90901 AS builder
```

Comment:
I think other repos have been `1.23.7`, is there a reason we had to stop at `1.23.6` in here?

Reply:
Since we're in here manually editing the files to go to `1.23`, I'm not really sure I understand why we'd knowingly bump to a version with a known vulnerability (`1.23.7` contains a security fix) though?

I guess you're saying that `dependabot` is just going to bump it to `1.24.1` immediately? But if that's what we're wanting to happen, then why not bump it to `1.24.1` ourselves in here?

Reply:
The rest of the `Dockerfile`s in this commit are all using `golang:1.23.6-alpine3.20`. This is why I bump the version to maintain the consistency. I can bump this to `1.24.1` for sure.

Reply:
As chatted with @AlCutter offline, bumping it to `1.24.1` saves us the time on the next version bump. It's fine to use the latest Go version in the examples repo.